File: 2022/06/ab.hs
-- Compile and run with:
--      $ ghc ab.hs; ./ab

import Data.List

main = do
    input <- readFile "input.txt"
    print $ findMarker input 4 0
    print $ findMarker input 14 0

findMarker (x:xs) size consumed
    | repeated (take size (x:xs)) = findMarker xs size (consumed+1)
    | otherwise = consumed + size
    where repeated xs = xs /= nub xs

File: 2022/06/ab.py
with open('input.txt') as file:
    input = file.read().strip()

def find_marker(input, size):
    for i in range(len(input)):
        if len(set(input[i:i+size])) == size:
            return i+size

print(find_marker(input, 4))
print(find_marker(input, 14))
